tasks/opensource/install-distribution.yml

```diff
@@ -32,13 +34,35 @@
     - ansible_facts['distribution_major_version'] is version('7', '==')
 
 - name: (Amazon Linux) Enable the NGINX Amazon extras package repository
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.command:
     cmd: amazon-linux-extras enable nginx1
     creates: /usr/sbin/nginx
-  when: ansible_facts['distribution'] == 'Amazon'
+  when:
+    - ansible_facts['distribution'] == 'Amazon'
+    - distribution != 'Amazon Linux 2023'
 
 - name: "{{ nginx_setup | capitalize }} NGINX from the distribution's package repository"
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.package:
     name: nginx{{ nginx_version | default('') }}
     state: "{{ nginx_state }}"
   notify: (Handler) Run NGINX
+  when:
+    - ansible_facts['distribution'] == 'Amazon'
+    - distribution != 'Amazon Linux 2023'
+
+- name: "{{ nginx_setup | capitalize }} NGINX from the distribution's package repository"
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
+  ansible.builtin.dnf:
+    name: nginx{{ nginx_version | default('') }}
+    disablerepo: amazonlinux
+    enablerepo: nginx-mainline-repo
+    state: "{{ nginx_state }}"
+  notify: (Handler) Run NGINX
+  when:
+    - ansible_facts['distribution'] == 'Amazon'
+    - distribution == 'Amazon Linux 2023'
```

Comment:
This task involves installing NGINX from the Amazon repo. I am pretty sure that's not what this block does? I am also not entirely sure if we even need an Amazon Linux 2023 specific task or whether NGINX is available in the core  Amazon Linux 2023 packages? (Judging by the fact you had to disable the amazonlinux repo I assume it is?)